app/routes/login.tsx

```diff
@@ -11,17 +11,25 @@ import {
 } from '@chakra-ui/react';
 import { LockIcon } from '@chakra-ui/icons';
 import type { ActionArgs } from '@remix-run/node';
+import { redirect } from '@remix-run/node';
 import { Form } from '@remix-run/react';
 
-import { createUserSession } from '~/session.server';
+import { getUsername } from '~/session.server';
+import { createLoginRequest } from '~/saml.server';
 
 export const action = async ({ request }: ActionArgs) => {
-  return createUserSession({
-    request: request,
-    username: 'starchartdev',
-    remember: false,
-    redirectTo: '/',
-  });
+  // Check if a session with a username exists
+  const user = await getUsername(request);
+
+  // If not then create a login request to the IDP's redirect binding
+  if (!user) {
+    const url = new URL(request.url);
+    const redirectTo = url.searchParams.get('redirectTo');
```

Comment:
```ts
const redirectTo = url.searchParams.get('redirectTo') ?? undefined;
```

This will give us `undefined` if we get back `null`, which is better for the `createLoginRequest` params.  I'll comment below on how to do it.